planner/core/pb_to_plan.go

```diff
@@ -0,0 +1,193 @@
+package core
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/expression/aggregation"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+type pbPlanBuilder struct {
+	sctx sessionctx.Context
+	tps  []*types.FieldType
+	is   infoschema.InfoSchema
+}
+
+// NewPBPlanBuilder creates a new pb plan builder.
+func NewPBPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema) *pbPlanBuilder {
```

Comment:
This exported function returns an unexported struct, I am not sure where is it a good definition. What do you think @djshow832 @bb7133